src/transform/src/column_knowledge.rs

```diff
@@ -701,11 +701,16 @@ impl DatumKnowledge {
                 unreachable!();
             };
 
-            if s_typ != o_typ {
+            if !s_typ.base_eq(o_typ) {
                 soft_panic_or_log!("Undefined meet of non-equal types {s_typ:?} != {o_typ:?}");
```

Comment:
Maybe you could update the msg from `!=` to `!base_eq`. (But anyhow when people look up this code after seeing the error msg, it will be clear what kind of equality is this, so it's not too important.)